### PR TITLE
Added `BaseInput` and `BaseIntent` for point-of-sale

### PR DESCRIPTION
**Pull request by fatbattk**
### Background

Closes https://github.com/Shopify/pos-next-react-native/issues/47313

### Solution

- Moved `api` and `components` folder under a new `render` folder.
- Moved `types` out from `api` folder to `point-of-sale`.
- Added a `run` folder with `input` and `intent` which has the base types.
- Added `device` type and added `currency` to `Session` type.

### 🎩

1) `yalc` to POS app and use the types there. `BaseInput` or `BaseIntent`.
1) On POS RN, `./scripts/ui-extensions/yalc-local-extensions add`

### Checklist

- [x] I have :tophat:'d these changes
- [ ] I have updated relevant documentation

### CODE REVIEW

File: packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  DeviceApiContent,
+  Session,
+} from '../../../point-of-sale';
+import type {CurrencyCode} from '../../../../shared';
+
+export interface BaseInput {
+  connectivity: ConnectivityState;
+  device: DeviceApiContent;
+  locale: string;
+  session: Session & {currency: CurrencyCode};
```
Comment (js-goupil) [inline packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts:12]: I think we should just include currency in the `Session` api. This is a wanted feature for UI rendering extensions as well!

File: packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import type {
+  ConnectivityState,
+  DeviceApiContent,
+  Session,
+} from '../../../point-of-sale';
+import type {CurrencyCode} from '../../../../shared';
+
+export interface BaseInput {
+  connectivity: ConnectivityState;
+  device: DeviceApiContent;
```
Comment (js-goupil) [inline packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts:10]: DeviceApiContent has a bunch of getter functions which are very "API-like". I think we should re-write this into an interface that just has read values, and POS can call the functions on its side and return the values there.

File: packages/ui-extensions/src/surfaces/point-of-sale/run/intent/BaseIntent.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+interface ExtensionError extends Error {
+  level: 'error' | 'warning';
```
Comment (js-goupil) [inline packages/ui-extensions/src/surfaces/point-of-sale/run/intent/BaseIntent.ts:2]: Wondering what the purpose of error vs warning will be on the POS side. Do we have a recommendation as to when either will be appropriate?

**Review by js-goupil — COMMENTED** [review]
Good start, but we should consolidate currencyCode into the Session interface. Also should avoid having method based APIs in our input

File: packages/ui-extensions/src/surfaces/point-of-sale/run/intent/BaseIntent.ts (hunk 3 above)
Comment (NathanJolly) [reply packages/ui-extensions/src/surfaces/point-of-sale/run/intent/BaseIntent.ts:2]: https://docs.google.com/document/d/1km0Oyz_QZTWUdFtTWAv3-1qmVNsaD_shY4VAmKwSZ2g/edit?tab=t.0#bookmark=id.39h1v6blsjs5 This is to trigger the yellow or red states. We can expound on it in docs etc

File: packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts (hunk 2 above)
Comment (fatbattk) [reply packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts]: added.

Comment (ProdigyXable) [reply packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts]: In the near future, different locations can have the locale/currency which may differ than what is set in the store's admin.

**Review by js-goupil — APPROVED** [review]

File: packages/ui-extensions/src/surfaces/point-of-sale/types/session.ts [hunk 4]
```diff
@@ -24,4 +26,9 @@ export interface Session {
    * Note that this staff member ID may be different to the User ID, as the staff member who enters their PIN may be different to the User who logged onto POS.
    */
   staffMemberId?: number;
+
+  /**
+   * The currency code associated with the location currently in on POS.
+   */
+  currency: CurrencyCode;
```
[the remark this answers is not in the document]
Comment (js-goupil) [reply packages/ui-extensions/src/surfaces/point-of-sale/types/session.ts:33]: I think as soon as we update on the POS side the linter will complain when it doesn't see this value getting passed in

**Pull request merged**